plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/DebugService.java

```diff
@@ -0,0 +1,216 @@
+/**
+ * Copyright (c) 2020 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.ide.xtext.server;
+
+import java.lang.management.LockInfo;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryMXBean;
+import java.lang.management.MemoryUsage;
+import java.lang.management.MonitorInfo;
+import java.lang.management.ThreadInfo;
+import java.lang.management.ThreadMXBean;
+import java.util.concurrent.CompletableFuture;
+
+import org.apache.log4j.Level;
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.apache.log4j.lf5.LogLevel;
+import org.eclipse.lsp4j.MessageParams;
+import org.eclipse.lsp4j.MessageType;
+import org.eclipse.lsp4j.jsonrpc.services.JsonRequest;
+import org.eclipse.lsp4j.jsonrpc.services.JsonSegment;
+import org.eclipse.lsp4j.services.LanguageClient;
+import org.eclipse.n4js.ide.xtext.server.DebugService.DebugServiceNullImpl;
+import org.eclipse.n4js.ide.xtext.server.build.ConcurrentIndex;
+import org.eclipse.n4js.utils.Strings;
+import org.eclipse.n4js.xtext.workspace.IWorkspaceConfigSnapshot;
+
+import com.google.inject.ImplementedBy;
+import com.google.inject.Inject;
+
+/**
+ * The {@link DebugService} uses a separate endpoint to avoid the infrastructure of other calls to ordinary LSP
+ * end-points and hence to increase robustness in case of errors in the source code.
+ */
+@JsonSegment("debug")
+@ImplementedBy(DebugServiceNullImpl.class)
+public interface DebugService {
+
+	/** Sets the client to communicate to */
+	void connect(LanguageClient client);
+
+	/** Sets the level of verbosity of Log4j. @see {@link LogLevel} */
+	@JsonRequest
+	CompletableFuture<Void> setVerboseLevel(String level);
+
+	/** Prints debug information on the output channel */
```

Comment:
I'm not sure about the wording.

The standard implementation will send debug information to the client via client.logMessage.

Question: Is `printDebugInfo` a request or a notification, given that we don't send the debug information back but notify the client about it?